gateway_lambda/types.ts

```diff
@@ -7,6 +7,7 @@ export type SqsEvent = {
   user_id: string;
   event: EVENT;
   timestamp: number;
+  userEmail?: string;
```

Comment:
This works, but it might be nice to have different types for different events, and differentiate them with `event`.

There's a real ambiguity here where you don't know if `userEmail` is optional here because it's not present on all event types, or if it's optional because even not all profile update events include an email change. Turns out it's the latter, but I have to check fxa's docs to know this. This problem compounds as more event types are added, and subscription and password change events are incoming soon.

Reply:
Yep, agree can refactor to have better TS types and comments and individual handler functions.